src/pages/Category/CategoryList.js

```diff
@@ -0,0 +1,146 @@
+  const dropBar = React.createRef();
+
+  const fetchCategoryData = () => {
+    fetch('/data/categoryList.json')
+      .then(res => res.json())
+      .then(data => {
+        setCategory(data.results);
+      });
+  };
+
+  useEffect(() => {
+    fetchCategoryData();
+    window.addEventListener('click', removeDropBarMenu);
+  }, []);
+
+  const handleDropBarMenu = e => {
+    const target = e.target.closest('button');
+    const targetData = category[target.id - 1].categoryList;
+    const targetLeft = target.getBoundingClientRect().left;
+    setCategoryList(targetData);
+    dropBar.current.style.left = `${targetLeft}px`;
+  };
+
+  const removeDropBarMenu = e => {
+    if (!e.target.matches('li')) {
+      setCategoryList([]);
+    }
+  };
+
+  const getFilterName = e => {
+    let value = e.target.id;
+    const filterName = categoryList[value - 1].categoryName;
+    setCategoryName(filterName);
+    console.log(filterName);
+  };
+
+  const handleFilterBtn = e => {
+    getFilterName(e);
+    props.getCategoryName(e);
+  };
+
+  return (
+    <>
+      <GlobalStyles />
+      <WrapCategory>
+        <FilterBar>
+          {category.map(el => {
+            return (
+              <DropDown key={el.id}>
+                <Button
+                  onClick={handleDropBarMenu}
+                  onMouseOver={handleDropBarMenu}
+                  id={el.id}
+                >
+                  <BtnText>{el.menuName}</BtnText>
+                  <FaChevronDown className="fas fa-chevron-down" />
+                </Button>
+              </DropDown>
+            );
+          })}
+          <DropBar ref={dropBar}>
+            {categoryList.map(el => {
+              return (
+                <DropBarList
+                  onClick={handleFilterBtn}
+                  key={el.categoryId}
+                  id={el.categoryId}
+                >
+                  {el.categoryName}
+                </DropBarList>
+              );
+            })}
+          </DropBar>
+        </FilterBar>
+        <ul>
+          <li>
+            <button>{categoryName}</button>
+          </li>
+        </ul>
+      </WrapCategory>
+    </>
+  );
+}
+
+const WrapCategory = styled.section`
+  width: 1415px;
+  margin: auto;
+  padding: 10px 0 5px;
+`;
+
+const FilterBar = styled.ul`
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+  padding: 5px 0;
+`;
+
+const DropDown = styled.li`
+  margin: 0 2px;
+`;
+
+const DropBar = styled.ul`
+  position: absolute;
+  top: 80px;
+  width: 210px;
+  background: #fff;
+  border-radius: 10px;
+  box-shadow: 0 0 1em 0.1px lightgray;
+  z-index: 1;
+`;
+
+const DropBarList = styled.li`
+  padding: 10px 20px;
+  line-height: 2;
+
+  &:hover {
+    background: rgba(7, 255, 255, 0.08);
+  }
+`;
+
+const Button = styled.button`
+  padding: 8px 10px 8px;
+  background-color: ${props => props.theme.backgroundGrey};
+  color: ${props => props.theme.fontFilterGray};
```

Comment:
```suggestion
  background-color: ${({theme}) => theme.backgroundGrey};
  color: ${({theme}) => theme.fontFilterGray};
```
매개변수부분에서 바로 구조분해할당 할 수 있습니다!

Reply:
구조분해할당은 다양한 곳에서 적용할 수 있군요! 수정했습니다!